File: src/Banco.java
import java.util.ArrayList;
import java.util.Scanner;

public class Banco {
    Scanner sc=new Scanner(System.in);
    private static ArrayList<Employee> empleados=new ArrayList<>();
    public ArrayList<Employee> getEmpleados() {
        return empleados;
    }
    public void agregarEmpleado(){
        System.out.println("Ingrese el nombre: ");
        String name = sc.next();
        System.out.println("Ingrese el apellido: ");
        String apellido = sc.next();
        System.out.println("Ingrese el numero de cuenta: ");
        long account = sc.nextLong();
        System.out.println("Ingrese el tipo de cuenta: ");
        char type = sc.next().charAt(0);
        empleados.add(new Employee(name, apellido, account, type));
    }
    public void eliminarEmpleado(Employee empleado){
        empleados.remove(empleado);
    }
    //muestra a todos los usuarios
    public void mostrarUsuarios(){
        if(empleados.isEmpty()){
            System.out.println("No hay usuarios registrados");
        }
        else {
            for (int i = 0; i < empleados.size(); i++) {
                mostrarUsuario(i);
            }
        }
    }
    //muestra un usuario en la posicion i
    public void mostrarUsuario(int i){
        System.out.printf("Nombre: %s\nApellido: %s", empleados.get(i).getName(), empleados.get(i).getLastName());
        System.out.print("Cuentas: ");
        for(int x=0;x<empleados.get(i).getAccounts().size();x++){
            System.out.println(empleados.get(i).getAccounts().get(x).getAccountNumber()+" ");
            System.out.printf("Dinero en cuenta: %f",empleados.get(i).getAccounts().get(x).getAmount());
        }
    }
    //buscador por nombre
    public void buscarNombre(String name){
        boolean band=true;
        for (int i = 0; i < empleados.size(); i++) {
            if(empleados.get(i).getName().equals(name)){
                mostrarUsuario(i);
                band=false;
            }
        }
        if(band){
            System.out.println("No se encontro este usuario: "+name);
        }
    }
    //buscador por apellido
    public void buscarApellido(String lastName){
        boolean band=true;
        for (int i = 0; i < empleados.size(); i++) {
            if(empleados.get(i).getLastName().equals(lastName)){
                mostrarUsuario(i);
                band=false;
            }
        }
        if(band){
            System.out.println("No se encontro este usuario: "+lastName);
        }
    }
    //buscador por numero de cuenta
    public void buscarCuenta(long account){
        boolean band=true;
        for (int i = 0; i < empleados.size(); i++) {
            for (int j = 0; j < empleados.get(i).getAccounts().size(); j++) {
                if(empleados.get(i).getAccounts().get(j).getAccountNumber()==account){
                    mostrarUsuario(i);
                    band=false;
                }
            }
        }
        if(band){
            System.out.println("No se encontro esta cuenta: "+account);
        }
    }
    public int[] posicionEmpleado(long account){
        int pos[]=new int[2];
        for (int i = 0; i < empleados.size(); i++) {
            for (int j = 0; j < empleados.get(i).getAccounts().size(); j++) {
                if (empleados.get(i).getAccounts().get(j).getAccountNumber() == account) {
                    pos[0] = i;
                    pos[1]=j;
                    break;
                }
            }
        }
        return pos;
    }
    public int posicionEmpleado(String name){
        int pos=-1;
        for (int i = 0; i < empleados.size(); i++) {
            for (int j = 0; j < empleados.get(i).getAccounts().size(); j++) {
                if (empleados.get(i).getName().equals(name)) {
                    pos = i;
                    break;
                }
            }
        }
        return pos;
    }
    public void buscarEmpleado(){
        System.out.println("¿Cómo desea buscar?\n1.Nombre\n2.Apellido\n3.Cuenta");
        int opcion = sc.nextInt();
        if (opcion == 1) {
            System.out.println("Ingrese el nombre: ");
            String name = sc.next();
            buscarNombre(name);

        }
        if (opcion == 2) {
            System.out.println("Ingrese el apellido: ");
            String apellido = sc.next();
            buscarApellido(apellido);
        }
        if (opcion == 3) {
            System.out.println("Ingrese el numero de cuenta: ");
            long account = sc.nextLong();
            buscarCuenta(account);
        }
    }
    public void agregarRetirar(){
        System.out.println("¿Desea agregar o retirar dinero?\n1.Agregar\n2.Retirar\n3.Volver al menu");
        int opcion=sc.nextInt();
        double cantidad;
        int posicion[]=new int[2];
        if(opcion==1){
            System.out.print("Ingrese la cuenta: ");
            long account= sc.nextLong();
            posicion=posicionEmpleado(account);
            System.out.println("¿Cuanto desea agregar?: ");
            cantidad=sc.nextDouble();
            empleados.get(posicion[0]).getAccounts().get(posicion[1]).agregarDinero(cantidad);
        }
        if(opcion==2) {
            System.out.print("Ingrese la cuenta: ");
            long account = sc.nextLong();
            posicion = posicionEmpleado(account);
            System.out.println("¿Cuanto desea agregar?: ");
            cantidad = sc.nextDouble();
            empleados.get(posicion[0]).getAccounts().get(posicion[1]).retirarDinero(cantidad);
        }
    }
    public void editarUsuarios(){
        System.out.println("¿Que desea hacer?: \n1.Agregar cuenta bancaria\n2.Eliminar usuario");
        int opcion=sc.nextInt();
        if(opcion==1){
            System.out.println("Ingrese el nombre de usuario:" );
            String name=sc.next();
            int pos=posicionEmpleado(name);
            System.out.println("Ingrese el numero de cuenta: ");
            long account=sc.nextLong();
            System.out.println("Ingrese el tipo de cuenta: ");
            char type=sc.next().charAt(0);
            empleados.get(pos).agregarCuenta(account,type);
        }
        if(opcion==2){
            System.out.println("Ingrese el nombre de usuario que desea eliminar: ");
            String name=sc.next();
            int pos=posicionEmpleado(name);
            Employee empleado=empleados.get(pos);
            eliminarEmpleado(empleado);
            System.out.printf("Empleado %s eliminado",name);
        }
    }
}

File: src/Banco.py
from Employee import Employee

class Banco:
    __empleados=[]
        
    def agregar_empleado(self):
        nombre=input("Ingrese el nombre: ")
        apellido=input("Ingrese el apellido: ")
        account=int(input("Ingrese el numero de cuenta: "))
        type=input("Ingrese el tipo de cuenta: ")[0]
        self.__empleados.append(Employee(nombre,apellido,account,type))
    
    def eliminar_empleado(self,empleado):
        self.__empleados.remove(empleado)
    
    def mostrar_usuarios(self):
        if len(self.__empleados)==0:
            print("No hay usuarios registrados")
        else:
            for i in range(0,len(self.__empleados)):
                self.mostrar_usuario(i)  
                
    def mostrar_usuario(self, entero):
        print("Nombre:",self.__empleados[entero].get_name(), "\nApellido:",self.__empleados[entero].get_last_name(),"\nCuentas:")
        for i in range(0,len(self.__empleados[entero].get_accounts())):
            print(self.__empleados[entero].get_accounts()[i].get_account_number())
            print("Dinero en cuenta: $",self.__empleados[entero].get_accounts()[i].get_amount())
    
    def buscar_nombre(self,nombre):
        band=True
        for i in range(0,len(self.__empleados)):
            if self.__empleados[i].get_name()==nombre:
                self.mostrar_usuario(i)
                band=False
        if band:
            print("No se encontro este usuario")
                
    def buscar_apellido(self,apellido):
        band=True
        for i in range(0,len(self.__empleados)):
            if self.__empleados[i].get_last_name()==apellido:
                self.mostrar_usuario(i)
                band=False
        if band:
            print("No se encontro este usuario")   
             
    def buscar_cuenta(self,cuenta):
        band=True
        for i in range(0,len(self.__empleados)):
            for j in range(0,len(self.__empleados[i].get_accounts())):
                self.mostrar_usuario(i)
                band=False
        if band:
            print("No se encontro la cuenta")
            
    def posicion_empleado_matriz(self,account):
        pos=[]
        for i in range(0,len(self.__empleados)):
            for j in range(0,len(self.__empleados[i].get_accounts())):
                if self.__empleados[i].get_accounts()[j].get_account_number()==account:
                    pos[0]=i
                    pos[1]=j
                    break
        return pos
    
    def posicion_empleado(self,name):
        pos=-1
        for i in range(0,len(self.__empleados)):
            for j in range(0,len(self.__empleados[i].get_accounts())):
                if self.__empleados[i].get_name()==name:
                    pos=i
                    break
        return pos  
    def buscar_empleado(self):
        opcion=int(input("Como desea buscar?\n1.Nombre\n2.Apellido\n3.Cuenta"))
        if opcion==1:
            name=input("Ingrese el nombre: ")
            self.buscar_nombre(name)
        elif opcion==2:
            apellido=input("Ingrese el nombre: ")
            self.buscar_apellido(apellido)  
        elif opcion==3:
            account=int(input("Ingrese el numero de cuenta: "))   
            self.buscar_cuenta(account)
            
    def agregar_retirar(self):
        opcion=int(input("¿De sea agregar o retirar dinero?\n1.Agregar\n2.Retirar\n3.Volver al menu"))
        if opcion==1:
            account=int(input("Ingrese la cuenta: "))
            posicion=self.posicion_empleado_matriz(account)
            cantidad=float(input("¿Cuanto desea agregar?"))
            self.__empleados[posicion[0]].get_accounts()[posicion[1]].agregar_dinero(cantidad)
        elif opcion==2:
            account=int(input("Ingrese la cuenta: "))
            posicion=self.posicion_empleado_matriz(account)
            cantidad=float(input("¿Cuanto desea retirar?"))
            self.__empleados[posicion[0]].get_accounts()[posicion[1]].retirar_dinero(cantidad)
        
    def editar_usuarios(self):
        opcion=int(input("¿Que desea hacer?: \n1.Agregar a cuenta bancaria\n2.Eliminar usuario"))
        if opcion==1:
            name=input("Ingrese el nombre del usuario")
            pos=self.posicion_empleado(name)
            account=int(input("Ingrese el numero de cuenta"))
            type=input("Ingrese el tipo de cuenta")[0]
            self.__empleados[pos].agregar_cuenta(account,type)
        elif opcion==2:         
            name=input("Ingrese el usuario que desea eliminar")
            pos=self.posicion_empleado(name)
            empleado=self.__empleados[pos]
            self.eliminar_empleado(empleado)
            print("Empleado: ",name," eliminado")